apps/backend/src/evaluations/evaluations.controller.ts

```diff
@@ -72,6 +72,15 @@ export class EvaluationsController {
       abac.can(Action.Read, evaluation)
     );
 
+    // Ensure user has the ability to read associated groups
+    evaluations.forEach((evaluation) => {
+      evaluation.groups = evaluation.groups.filter(
+        (group) =>
+          abac.can(Action.AddEvaluation, group) &&
+          abac.can(Action.RemoveEvaluation, group)
```

Comment:
Shouldn't this be `Action.Read, group`?

Reply:
It seems that Action.Read doesn't filter out access to groups, which groupsForEvaluation seems to avoid as well.

Reply:
If it doesn't it sounds like a bug, I'd rather fix that if it is the case.

Reply:
The tests were assuming that public groups could not be read, which I don't think is the correct way to go about things. I've changed it to check Action.Read instead for both `groupsForEvaluation` and `findAll`.